packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -0,0 +1,343 @@
+import { FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import type { BTCSignTx, SolanaSignTx } from '@shapeshiftoss/hdwallet-core'
+import type { EvmChainId, KnownChainIds } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { AxiosError } from 'axios'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import type {
+  EvmTransactionRequest,
+  GetUnsignedEvmTransactionArgs,
+  GetUnsignedSolanaTransactionArgs,
+  GetUnsignedUtxoTransactionArgs,
+  SwapperApi,
+  UtxoFeeData,
+} from '../../types'
+import { isExecutableTradeQuote, isExecutableTradeStep, isToken } from '../../utils'
+import { CHAINFLIP_BAAS_COMMISSION, chainIdToChainflipNetwork } from './constants'
+import type { ChainflipBaasSwapDepositAddress } from './models/ChainflipBaasSwapDepositAddress'
+import { getTradeQuote, getTradeRate } from './swapperApi/getTradeQuote'
+import type { ChainFlipStatus } from './types'
+import { chainflipService } from './utils/chainflipService'
+import { getLatestChainflipStatusMessage } from './utils/getLatestChainflipStatusMessage'
+
+// Persists the ID so we can look it up later when checking the status
+const tradeQuoteMetadata: Map<string, ChainflipBaasSwapDepositAddress> = new Map()
+
+export const chainflipApi: SwapperApi = {
+  getTradeQuote,
+  getTradeRate,
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    tradeQuote,
+    assertGetEvmChainAdapter,
+    config,
+    supportsEIP1559,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const brokerUrl = config.REACT_APP_CHAINFLIP_API_URL
+    const apiKey = config.REACT_APP_CHAINFLIP_API_KEY
+
+    const step = tradeQuote.steps[0]
+    const sellChainflipChainKey = `${step.sellAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.sellAsset.chainId]
+    }`
+    const buyChainflipChainKey = `${step.buyAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.buyAsset.chainId]
+    }`
+
+    // Subtract the BaaS fee to end up at the final displayed commissionBps
+    let serviceCommission = parseInt(tradeQuote.affiliateBps) - CHAINFLIP_BAAS_COMMISSION
+    if (serviceCommission < 0) serviceCommission = 0
+
+    const maybeSwapResponse = await chainflipService.get<ChainflipBaasSwapDepositAddress>(
+      `${brokerUrl}/swap` +
+        `?apiKey=${apiKey}` +
+        `&sourceAsset=${sellChainflipChainKey}` +
+        `&destinationAsset=${buyChainflipChainKey}` +
+        `&destinationAddress=${tradeQuote.receiveAddress}` +
+        `&boostFee=10` +
+        // TODO: Calculate minprice based on tradeQuote.slippageTolerancePercentageDecimal, step.sellAmountIncludingProtocolFeesCryptoBaseUnit, step.buyAmountAfterFeesCryptoBaseUnit
+        // `&minimumPrice=` +
+        // `&refundAddress=${from}` +
+        // `&retryDurationInBlocks=10` +
```

Comment:
@CumpsD is this still a valid TODO ? Does this parameter act as a limit to ensure users aren't rugged and losing funds?

Reply:
@gomesalexandre yes, still valid! This is slippage protection, which indeed ensures users arent rugged and paying way too much on a swap.

This should be the calculated using the value from the UI:
![image](https://github.com/user-attachments/assets/e63df685-26c5-4f1f-9ca3-7710c6499b59)

Reply:
From the docs:

> A minimum accepted price is specified as the ratio between the human readable destination amount and the human readable source amount e.g. 3500 for a swap from ETH to USDC if you want to receive at least 3500 USDC per ETH. Prices are checked on the AMM level and do not include deposit or broadcast fees. To calculate a minPrice based on a [quote](https://docs.chainflip.io/swapping/integrations/javascript-sdk/swap-assets/get-quote), the estimatedPrice value is multiplied with the desired slippage tolerance. For example, quote.estimatedPrice * 99 / 100 specifies a 1% slippage tolerance relative to the quote.

See more details and examples here: https://docs.chainflip.io/swapping/integrations/javascript-sdk/swap-assets/request-deposit-address/v1#fillorkillparams-slippage-protection

Reply:
Tyvm @CumpsD! Done in 5da50735d021e0383b2168e0233bf60753116271, added `minimumPrice` in alongside required `refundAddress` and `retryDurationInBlocks` as refund params. 

<img width="596" alt="Screenshot 2024-11-12 at 18 46 18" src="https://github.com/user-attachments/assets/d029dacb-4969-40e6-999a-b73a65e29f72">
<img width="244" alt="Screenshot 2024-11-12 at 18 46 41" src="https://github.com/user-attachments/assets/a629f7ef-a1cb-4382-8538-be7d4bfd0d1b">
<img width="596" alt="image" src="https://github.com/user-attachments/assets/603a70ca-aacf-425d-877f-e53e9f52657c">

<img width="587" alt="image" src="https://github.com/user-attachments/assets/8e30f164-5f8e-455c-91ef-2e614077114e">
<img width="208" alt="image" src="https://github.com/user-attachments/assets/e1302e27-6975-4003-b3ec-b3d8ab9acd28">
<img width="591" alt="image" src="https://github.com/user-attachments/assets/5b1f6020-6c2d-4a45-aea9-4decc0024836">